### ISSUE TITLE
Generic share API to share files with apps

### ISSUE THREAD

**Issue by juliushaertl** [nextcloud/server#10292]
We are adding more and more share types to the server component for apps that need to share files depending on the apps data and ownership model. 

Right now we have the following ones, that I think could also be moved to a common apps share:

- SHARE_TYPE_GUEST
- SHARE_TYPE_CIRCLE
- SHARE_TYPE_ROOM

As it would also make sense for other apps like calendar, tasks or deck to access files and share them with users inside there, it would make sense to think about a common share type for apps.

Permission handling would need to be done by the app somehow. Should files that a user got access to by sharing though apps be visible in the file tree? If not, how can users access/download/edit it?

@nextcloud/sharing

**Comment by juliushaertl** [comment]
cc @danxuliu @daita since you are already integrating the file sharing with an additional app.

**Comment by skjnldsv** [comment]
@daita this is what your Entity pull request would implement, right?

**Comment by daita** [comment]
@skjnldsv: yes, this is one of the thing that would be fixed by Entity

**Comment by szaimen** [comment]
I suppose this issue is still valid? If not, please close this issue!

**Comment by juliushaertl** [comment]
Would that be something covered by the planned circles work for 23 in server @ArtificialOwl ?

**Comment by ArtificialOwl** [comment]
> Would that be something covered by the planned circles work for 23 in server @ArtificialOwl ? I remember you mentioned app circle types

Yes, well not using by App SingleIds as they will be used if an app wants to generate its own grouping of users.
But yes, the future of Circles would be to replace all Share Types
